graphql_compiler/compiler/emit_sql.py

```diff
@@ -88,27 +100,187 @@ def _create_query(node, context):
+        expressions.BinaryComposition: _transform_binary_composition_to_expression,
+    }
+    expression_type = type(expression)
+    if expression_type not in _expression_transformers:
+        raise NotImplementedError(
+            u'Unsupported compiler expression "{}" of type "{}" cannot be converted to SQL '
+            u'expression.'.format(expression, type(expression)))
+    return _expression_transformers[expression_type](expression, node, context)
+
+
+def _transform_binary_composition_to_expression(expression, node, context):
+    """Transform a BinaryComposition compiler expression into a SQLAlchemy expression.
+
+    Recursively calls _expression_to_sql to convert its left and right sub-expressions.
+
+    Args:
+        expression: expression, BinaryComposition compiler expression.
+        node: SqlNode, the SqlNode the expression applies to.
+        context: CompilationContext, global compilation state and metadata.
+
+    Returns:
+        Expression, SQLAlchemy expression.
+    """
+    if expression.operator not in constants.SUPPORTED_OPERATORS:
+        raise NotImplementedError(
+            u'Filter operation "{}" is not supported by the SQL backend.'.format(
+                expression.operator))
+    sql_operator = constants.SUPPORTED_OPERATORS[expression.operator]
+    left = _expression_to_sql(expression.left, node, context)
+    right = _expression_to_sql(expression.right, node, context)
+    if sql_operator.cardinality == constants.CARDINALITY_UNARY:
+        left, right = _get_column_and_bindparam(left, right, sql_operator)
+        clause = getattr(left, sql_operator.name)(right)
+        return clause
+    elif sql_operator.cardinality == constants.CARDINALITY_BINARY:
+        clause = getattr(sql_expressions, sql_operator.name)(left, right)
+        return clause
+    elif sql_operator.cardinality == constants.CARDINALITY_LIST_VALUED:
+        left, right = _get_column_and_bindparam(left, right, sql_operator)
+        # ensure that SQLAlchemy treats the right bind parameter as list valued
+        right.expanding = True
+        clause = getattr(left, sql_operator.name)(right)
+        return clause
+    else:
+        raise AssertionError(u'Unknown operator cardinality {}'.format(sql_operator.cardinality))
+
+
+def _get_column_and_bindparam(left, right, operator):
+    """Return left and right expressions in (Column, BindParameter) order."""
+    if not isinstance(left, Column):
+        left, right = right, left
+    if not isinstance(left, Column):
+        raise AssertionError(
+            u'SQLAlchemy operator {} expects Column as left side the of expression, got {} '
+            u'of type {}'.format(operator, left, type(left)))
+    if not isinstance(right, BindParameter):
+        raise AssertionError(
+            u'SQLAlchemy operator {} expects BindParameter as the right side of the expression, got'
+            u' {} of type {}'.format(operator, right, type(right)))
```

Comment:
nit: Can you push the whitespace to the previous line? It kind of looks awkward to start a string literal with a space.

Reply:
Sure, yeah it does look a little weird.